### PR TITLE
Updated user registration tests after feedback

### PR DESCRIPTION
**Pull request by shenoac**
Added  User Registration Tests
-Duplicate Email:
Attempt to register with an existing email should return a 409 status with the message "Email is already taken".
-Invalid Data:
Missing name, email, or password should return a 400 status with a validation error message.
Invalid email format or password length should also fail with 400.

After feedback from first pr:
-removed console logs
-only pushed the test file
-removed password and email
-added a variable for the register endpoint

### CODE REVIEW

File: src/tests/user.test.ts [hunk 1]
```diff
@@ -1,25 +1,88 @@
...
 
 describe("User Tests", () => {
     const randomUser = {
         email: faker.internet.email(),
         password: faker.internet.password(),
         name: faker.person.fullName()
-    }
+    };
+
+    const registerEndpoint = '/api/users/register';
```
Comment (ahzia) [inline src/tests/user.test.ts:21]: Good job!

File: src/tests/user.test.ts [hunk 2]
```diff
@@ -1,25 +1,88 @@
...
+        const res = await request(app).post(registerEndpoint).send(randomUser);
+        expect(res.status).toBe(409);
+        expect(res.body.message).toBe("Email is already taken");
+    });
+
+    it("should return 400 for a missing name", async () => {
+        const invalidUser = {
+            email: faker.internet.email(),
+            password: faker.internet.password(),
+            name: ""
```
Comment (ahzia) [inline src/tests/user.test.ts:40]: Remove this line

**Comment by shenoac** [comment]
Converting to a draft while preparing for the review

**Comment by shenoac** [comment]
Hello @ahzia all the tests have passed ion the workflow, now ready for your review

File: src/tests/user/register.test.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      name: faker.person.fullName(),
+    };
+    const res = await request(app).post(registerEndpoint).send(invalidUser);
+    expect(res.status).toBe(400);
+    expect(res.body.error).toContain(
+      '"password" length must be at least 6 characters long',
+    );
+  });
+
+  describe('user login tests', () => {
```
Comment (ahzia) [inline src/tests/user/register.test.ts:90]: @shenoac can you remove the login tests from this file, as we moved it to the other file?

Comment (shenoac) [reply src/tests/user/register.test.ts]: done

**Review by ahzia — APPROVED** [review]
Thanks @shenoac you can merge your code!

**Pull request merged**